packages/cli-kit/src/ui/executor.test.ts

```diff
@@ -0,0 +1,81 @@
+import {groupAndMapChoices} from './executor.js'
+import {it, describe, expect} from 'vitest'
+
+describe('groupAndMapChoices', () => {
+  it('return choices ordered by name and without separator when group information is absent', async () => {
+    const choice1 = {
+      name: 'choice2',
+      value: 'value2',
+    }
+    const choice2 = {
+      name: 'choice1',
+      value: 'value1',
+    }
+
+    const result = groupAndMapChoices([choice1, choice2])
+    expect(result).toHaveLength(2)
+    expect((result[0] as {name: string; value: string}).name).toEqual('choice1')
+    expect((result[1] as {name: string; value: string}).name).toEqual('choice2')
```

Comment:
Nitpick: I'd recommend grouping the steps with:
```
// Given

// When

// Then
```

It's a weak convention that makes it easier to parse  scenarios' steps.